### ISSUE TITLE
Algorithm is not truly shortest path.

### ISSUE THREAD

**Issue by gonzagab** [kbumsik/WolfieMouse#26]
When the goal is surrounded by cells that have not been searched yet, the mouse will not actually take the shortest. The mouse will move to the left until it cannot. Then up and then right towards the goal.
```
 __ __ __ __ __ __ __ __ __ __ __ __ __ __ __ __ 
|  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  |
 ** ** ** ** ** ** ** ** ** ** ** ** ** ** ** ** 
|  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * G|
 ** ** ** ** ** ** ** __ ** ** ** ** ** ** ** ** 
|  *  *  *  *  *  *  |  |  *  *  *  *  *  *  *  |
 __ __ __ __ __ __ __ .. ** ** ** ** ** ** ** ** 
|  .  .  .  .  .  .  .  |  *  *  *  *  *  *  *  |
 .. .. __ __ __ __ __ __ __ ** ** ** ** ** ** ** 
|  |  .  .  .  .  .  .  |  |  *  *  *  *  *  *  |
 .. .. __ __ __ __ __ __ .. ** ** ** ** ** ** ** 
|  |  .  .  .  .  .  .  .  |  *  *  *  *  *  *  |
 .. .. __ __ __ __ __ __ .. ** ** ** ** ** ** ** 
|  |  |  *  *  *  *  *  |  |  *  *  *  *  *  *  |
 .. .. ** ** ** ** ** __ .. ** ** ** ** ** ** ** 
|  |  |  *  *  *  *  |  .  |  *  *  *  *  *  *  |
 .. .. ** ** ** __ __ .. .. __ __ __ .. ** ** ** 
|  |  |  *  *  |  .  .  .  |  .  .  .  |  *  *  |
 .. .. __ __ __ .. __ __ __ __ __ __ .. ** ** ** 
|  |  .  .  .  |  .  .  .  .  |  .  |  |  *  *  |
 .. __ __ __ .. .. __ __ __ .. __ __ .. ** ** ** 
|  .  .  .  |  |  |  .  .  |  .  .  .  |  *  *  |
 __ __ __ .. .. .. __ __ __ .. __ __ .. ** ** ** 
|  .  .  .  |  |  .  .  .  .  .  .  |  |  *  *  |
 .. __ __ __ .. __ __ __ __ __ __ __ .. ** ** ** 
|  .  .  .  |  .  .  .  .  .  .  .  .  |  *  *  |
 __ __ __ __ .. __ __ __ __ __ __ __ __ ** ** ** 
|  .  .  .  .  |  *  *  *  *  |  .  |  *  *  *  |
 .. .. __ __ __ ** ** ** ** ** .. .. ** ** ** ** 
|  |  |  *  *  *  *  *  *  *  |  |  |  *  *  *  |
 .. .. __ __ __ __ __ __ __ __ .. __ ** ** ** ** 
| S|  .  .  .  .  .  .  .  .  .  |  *  *  *  *  |
 __ __ __ __ __ __ __ __ __ __ __ __ __ __ __ __ 
```

**Comment by gonzagab** [comment]
```
| x* x* x* x* x* x* x* x* x*09*08*07*08*09* x* x|
 ** ** ** ** ** ** ** ** ** ** ** ** ** ** ** **
| x* x* x* x* x* x* x* x*09*08*07*06*07*08*09* x|
 ** ** ** ** ** ** ** __ ** ** ** ** ** ** ** **
| x* x* x* x* x* x* x| x|08*07*06*05*06*07*08* G|
 __ __ __ __ __ __ __ .. ** ** ** ** ** ** ** **
| x. x. x. x. x. x. x. x|07*06*05*04*05*06*07*08|
 .. .. __ __ __ __ __ __ __ ** ** ** ** ** ** **
| x| x. x. x. x. x. x. x| x|05*04*03*04*05*06*07|
 .. .. __ __ __ __ __ __ .. ** ** ** ** ** ** **
| x| x. x. x. x. x. x. x. x|04*03*02*03*04*05*06|
 .. .. __ __ __ __ __ __ .. ** ** ** ** ** ** **
| x| x| x* x* x* x* x* x| x|03*02*01*02*03*04*05|
 .. .. ** ** ** ** ** __ .. ** ** ** ** ** ** **
| x| x| x* x* x* x* x| x. x|02*01* M*01*02*03*04|
 .. .. ** ** ** __ __ .. .. __ __ __ .. ** ** **
| x| x| x* x* x| x. x. x. x|05.04.03.02|03*04*05|
 .. .. __ __ __ .. __ __ __ __ __ __ .. ** ** **
| x| x. x. x. x| x. x. x.09.08| x. x|03|04*05*06|
 .. __ __ __ .. .. __ __ __ .. __ __ .. ** ** **
| x. x. x. x| x| x| x. x. x|07.06.05.04|05*06*07|
 __ __ __ .. .. .. __ __ __ .. __ __ .. ** ** **
| x. x. x. x| x| x. x. x.09.08.09. x|05|06*07*08|
 .. __ __ __ .. __ __ __ __ __ __ __ .. ** ** **
| x. x. x. x| x. x. x. x. x.09.08.07.06|07*08*09|
 __ __ __ __ .. __ __ __ __ __ __ __ __ ** ** **
| x. x. x. x. x| x* x* x* x* x| x. x|09*08*09* x|
 .. .. __ __ __ ** ** ** ** ** .. .. ** ** ** **
| x| x| x* x* x* x* x* x* x* x| x| x| x*09* x* x|
 .. .. __ __ __ __ __ __ __ __ .. __ ** ** ** **
| S| x. x. x. x. x. x. x. x. x. x| x* x* x* x* x|
 __ __ __ __ __ __ __ __ __ __ __ __ __ __ __ __
```
pathStack: (7,11) (7,10) (7,9) (6,9) (6,10) (6,11) (6,10) (6,9) (5,9) (5,10) (5,11) (5,10) (5,9) (4,9) (4,10) (4,11) (4,10) (4,9) (3,9) (3,10) (3,11) (3,10) (3,9) (3,8) (2,8) (2,9) (2,10) (2,11) (2,10) (2,9) (1,9) (1,10) (1,11) (1,10) (0,10) (0,11) (0,12) (1,12) (2,12) (1,12) (1,13) (2,13) (2,14)

**Comment by gonzagab** [comment]
Since it is free space then there are multiple distances that are currentPosition + 1. So we need to implement some kind of dept first search algorithm. but this leads to a problem. we are forced to use dynamic memory allocation (e.g.`malloc`) to implement tree data structure. 

We could make tree data structure by using static array, however, the size of array could be huge. We need to examine how big the array will be.

**Comment by kbumsik** [comment]
Good. This issue is resolved in #30. Close issue.
